File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline

N = int(input())
queue = []
for _ in range(N):
    cmd = input().split()

    if cmd[0] == 'push':
        queue.append(cmd[1])
    elif cmd[0] == 'pop':
        print(queue.pop(0)) if len(queue) > 0 else print('-1')
    elif cmd[0] == 'size':
        print(len(queue))
    elif cmd[0] == 'empty':
        print('1') if len(queue) == 0 else print('0')
    elif cmd[0] == 'front':
        print(queue[0]) if len(queue) > 0 else print('-1')
    elif cmd[0] == 'back':
        print(queue[-1]) if len(queue) > 0 else print('-1')

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.StringTokenizer;

public class Main {
    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static StringTokenizer tokens;
    private static StringBuilder sb = new StringBuilder();

    private static int N;

    public static void main(String[] args) throws IOException {
        N = Integer.parseInt(br.readLine());
        ArrayDeque<Integer> q = new ArrayDeque<>();

        for (int i = 0; i < N; i++) {
            String[] tokens = br.readLine().split(" ");

            if (tokens[0].equals("push")) {
                q.addLast(Integer.parseInt(tokens[1]));
            } else if (tokens[0].equals("pop")) {
                sb.append(!q.isEmpty() ? q.pollFirst() : -1).append("\n");
            } else if (tokens[0].equals("size")) {
                sb.append(q.size()).append("\n");
            } else if (tokens[0].equals("empty")) {
                sb.append(q.isEmpty() ? 1 : 0).append("\n");
            } else if (tokens[0].equals("front")) {
                sb.append(!q.isEmpty() ? q.peekFirst() : -1).append("\n");
            } else if (tokens[0].equals("back")) {
                sb.append(!q.isEmpty() ? q.peekLast() : -1).append("\n");
            }
        }

        System.out.println(sb);
    }
}